### ISSUE TITLE
dotnet45 installation halts on bug 34803 workaround

### ISSUE THREAD

**Issue by IvanDSM** [Winetricks/winetricks#808]
On both winetricks 20170517 and 20170517-next (sha256sum: 06062b1c8e561fed5223c09cd3031252e02c313d3b8b89bef213a771d691f3d2) with Wine 2.9 Staging, the installation process for dotnet45 halts at the bug 34803 workaround. Here is the terminal output of winetricks dotnet45:

```$ winetricks dotnet45
Using winetricks 20170517 - sha256sum: 0461b25e833fd242b65addc72a5195c0047d3aaa91136b768d37102acb829a8e with wine-2.9 (Staging) and WINEARCH=win32
Executing w_do_call dotnet45
Executing load_dotnet45 
Executing w_do_call remove_mono
Executing load_remove_mono 
------------------------------------------------------
Mono does not appear to be installed.
------------------------------------------------------
------------------------------------------------------
Working around wine bug 34803 
------------------------------------------------------
reg: The system was unable to find the specified registry key or value
reg: The system was unable to find the specified registry key or value
reg: The system was unable to find the specified registry key or value
Executing rm -f /home/ivan/.wine/dosdevices/c:/windows/system32/mscoree.dll
Executing w_do_call dotnet40
Executing load_dotnet40 
------------------------------------------------------
dotnet40 does not yet fully work or install on wine.  Caveat emptor.
------------------------------------------------------
Executing w_do_call remove_mono
Executing load_remove_mono 
------------------------------------------------------
Mono does not appear to be installed.
------------------------------------------------------
------------------------------------------------------
Working around wine bug 34803 
------------------------------------------------------
reg: The system was unable to find the specified registry key or value
reg: The system was unable to find the specified registry key or value
reg: The system was unable to find the specified registry key or value
Executing rm -f /home/ivan/.wine/dosdevices/c:/windows/system32/mscoree.dll
Executing w_do_call winxp
Executing load_winxp 
Setting Windows version to winxp
Executing winetricks_early_wine regedit C:\windows\Temp\_winxp\set-winver.reg

```
After winetricks prints out "`Executing winetricks_early_wine regedit C:\windows\Temp\_winxp\set-winver.reg`" it just does nothing until I hit Ctrl+C.

**Comment by yacn** [comment]
I'm having the same trouble trying to install `dotnet35`/`dotnet35sp1`, it stops doing anything after "wine bug 34803" until killed:

```
------------------------------------------------------
Working around wine bug 34803 
------------------------------------------------------
The operation completed successfully
reg: The system was unable to find the specified registry key or value
reg: The system was unable to find the specified registry key or value
Executing rm -f /home/isaac/.vsphere-client//dosdevices/c:/windows/system32/mscoree.dll
Executing w_do_call dotnet20
dotnet20 already installed, skipping
------------------------------------------------------
Setting windows version so installer works
------------------------------------------------------
Setting Windows version to win2k
Executing winetricks_early_wine regedit C:\windows\Temp\_dotnet20sp1\set-winver.reg
^C
```

This is with winetricks 20170517 and Wine 2.10.

**Comment by vlovich** [comment]
My issue was that cabextract wasn't linked by homebrew (found by running winetricks with -v) & it failed with a similar error.  Is that your issue?  If so, winetricks should probably add an explicit check for any non-default system binaries it expects to be installed for that verb so that the error message is more helpful.

**Comment by austin987** [comment]
What do you mean "linked by"?

**Comment by austin987** [comment]
I'm still curious about @vlovich comment regarding homebrew and 'linked by', but dotnet45's cabextract was indeed broken, fixed by https://github.com/Winetricks/winetricks/commit/11d778fb69f9ea1e88a96948544161bbdb17078d

**Comment by Ghosy** [comment]
I'm currently experiencing the same issue while trying to install dotnet35.

winetricks 20170614-next
wine-2.12

**Comment by tjackson** [comment]
On a 64-bit prefix wine bug 34803 strikes and no version of dotnet4* can be installed (installer says higher version already installed) unless one moves away .cache/wine/wine-mono*, removes the system provided version, or cancels the download when creating the prefix. In a way that makes sense, but doesn't seem to be an issue on 32-bit prefixes.

That was my finding and solution on wine-staging 2.14 anyway.

**Comment by austin987** [comment]
@tjackson that was very helpful, thank you. I can reproduce in a 64-bit prefix. It appears to be a separate issue, but still good to know.

**Comment by austin987** [comment]
https://github.com/austin987/winetricks/tree/dotnet45-64

Fixes the issue with mono, which should allow 64-bit .Net to at least attempt to install. Dotnet45 installs for me, but i haven't tested all version yet.

I'd appreciate people testing it.

**Comment by tjackson** [comment]
Hey, got a chance to test this branch on wine-staging 2.18, 64-bit prefix.

As an aside, winetricks dotnet40 also mentions the unimplemented function winrt-roparameterizediid, but I think that actually isn't an issue until dotnet452.

This branch allowed me to use winetricks dotnet40 for my first test, installing Smite without having to avoid installing Mono, so it fixed the issue I described here a few comments above.

dotnet452 and up are still very difficult to install, randomly hanging in a critical section (ntdll / virtual.c), and I couldn't get Magic the Gathering Online to recognize an installed .NET version (tries to install its own 4.52 but hangs) on 64-bit. It detects our .NET, installs and runs on a 32-bit prefix though.

**Comment by austin987** [comment]
Thanks. I have some rebasing to do, but will merge soon since it improves things

**Comment by austin987** [comment]
@tjackson that branch is now merged, thanks for testing

**Comment by tjackson** [comment]
Great, thanks. It definitely does improve things. Also tested the Epic Games Launcher installation with success via dotnet40.

**Comment by austin987** [comment]
Great!. I'm looking at the rest of win64 as time permits.

**Comment by vlovich** [comment]
@austin987, sorry for the delayed reply.  Cabextract needed to be fixed by running brew link to fix the stmlinks. Seems like it wasn’t symlinked properly.

**Comment by yhcheng888** [comment]
with wine 5.0 in work
$ export WINEARCH=win64 && export WINEPREFIX="/home/cheng888/.wine64w" && wineboot --init
111 $ winecfg  - - - > change OS to Windows xp (must xp)
111 $ winetricks --force dotnet40  ( long wait )

$ winecfg  - - - > change OS to Windows 2008 (must 2008, or one can try other win-version)
222 $ wine64 /path to your file/LinuxMap5/NDP452-KB2901907-x86-x64-AllOS-ENU.exe
e.x.
222 $ wine64 /media/cheng888/Sp70c1/LinuxMap5/NDP452-KB2901907-x86-x64-AllOS-ENU.exe (win 2008 ,  can install only once)

$ winetricks --force mfc42 msls31 msvcirt msxml4 msxml6 vb6run vcrun6  (choose those you need)

execution example:
$ winecfg  - - - > Windows 2008 ( !!!! important )
$ wine64 explorer /desktop=name,7500x2100 "C:\yuantax\yeswin\yeswin.exe" 
  (this program needs dotnet40 and dotnet452, with two 43'' LCDs under linux cinnamon 19.3)


- - - - - - - if one needs to reinstall dotnet452, must reinstall wine5.0
to reinstall NDP452-KB2901907-x86-x64-AllOS-ENU.exe
$ sudo apt purge --auto-remove winehq-stable
$ sudo apt update && sudo apt install --install-recommends winehq-stable
$ sudo apt install winetricks

- - - - - - - 
------- if first time to install wine5.0, do the following first before installing:
download  libfaudio0_19.07-0_bionic_amd64.deb
copy to 
/var/cache/apt/archives/libfaudio0_19.07-0_bionic_amd64.deb
download  libfaudio0_19.07-0_bionic_i386.deb
copy to 
/var/cache/apt/archives/libfaudio0_19.07-0_bionic_i386.deb

sudo dpkg --add-architecture i386
sudo dpkg -i /var/cache/apt/archives/libfaudio0_19.07-0_bionic_amd64.deb   (a must for wine 5.0)
sudo dpkg -i /var/cache/apt/archives/libfaudio0_19.07-0_bionic_i386.deb   (a must for wine 5.0)
---------

**Comment by yhcheng888** [comment]
sudo dpkg --add-architecture i386
wget -nc https://dl.winehq.org/wine-builds/winehq.key
sudo apt-key add winehq.key
sudo apt install -f  (-f = fix dependency, if necessary)
sudo apt-add-repository 'deb https://dl.winehq.org/wine-builds/ubuntu/ bionic main'
sudo apt update
sudo apt install --install-recommends winehq-stable
sudo apt install winetricks
winetricks --force mfc42 msls31 msvcirt msxml4 msxml6 vb6run vcrun6 (choose those you need)

**Comment by seisdr** [comment]
```
Executing cd /root
Executing w_do_call remove_mono
Executing mkdir -p /root
Executing load_remove_mono
0134:fixme:file:NtLockFile I/O completion on lock not implemented yet
0134:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0154:fixme:file:NtLockFile I/O completion on lock not implemented yet
0154:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0154:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
0154:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
0134:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
0134:fixme:msi:internal_ui_handler internal UI not implemented for message 0x0b000000 (UI level = 5)
uninstaller: The application with GUID '{DE030A38-80E5-5F9E-9E66-115DF92F4865}' was not found
------------------------------------------------------
Working around wine bug 34803
------------------------------------------------------
reg: The system was unable to find the specified registry key or value
reg: The system was unable to find the specified registry key or value
reg: The system was unable to find the specified registry key or value
Executing rm -f /root/.wine/dosdevices/c:/windows/system32/mscoree.dll
Executing w_do_call dotnet40
Executing mkdir -p /root
Executing load_dotnet40
------------------------------------------------------
dotnet40 does not yet fully work or install on wine.  Caveat emptor.
------------------------------------------------------
Executing mkdir -p /root/.cache/winetricks/dotnet40
Executing cd /root/.cache/winetricks/dotnet40
Downloading https://download.microsoft.com/download/9/5/A/95A9616B-7A37-4AF6-BC36-D6EA96C8DAAE/dotNetFx40_Full_x86_x64.exe to /root/.cache/winetricks/dotnet40
```

**Comment by seisdr** [comment]
```
Using winetricks 20200412-next - sha256sum: d3dc42fb30624c96df880596a20f924aa90690b2126a9546e91088809472cb9c with wine-5.14 (Staging) and WINEARCH=win32
```

**Comment by seisdr** [comment]
```
03d4:err:eventlog:ReportEventW L".NET Runtime version 2.0.50727.42 - Fatal Execution Engine Error (7A05E2B3) (80131506)"
03d4:fixme:advapi:DeregisterEventSource (0xcafe4242) stub                               03d4:err:virtual:virtual_setup_exception stack overflow 860 bytes in thread 03d4 addr 0x9ec0d676 stack 0x220ca4 (0x220000-0x221000-0x320000)
```

**Comment by tjackson** [comment]
Github seems to have summoned me, so I'll just chime in and say that I tested this just now and can't reproduce a problem. You really shouldn't run Wine as root, and maybe that's the source of your problem?

**Comment by seisdr** [comment]
I get always stackoverflow on various application I'm not running as root

**Comment by seisdr** [comment]
and it really annoying me
here's when it try to execute this command the error occurs and freeze 
![IMG_٢٠٢٠٠٨١١_٠٦٤٣١٥](https://user-images.githubusercontent.com/45742462/89855094-39d81680-db9e-11ea-8a46-081d881f71a4.png)

**Comment by austin987** [comment]
A) this issue is closed
B) Sounds like https://bugs.winehq.org/show_bug.cgi?id=49690. I only get that with a 64-bit prefix, though I haven't tested with staging.
